components/compositing/windowing.rs

```diff
@@ -105,6 +106,10 @@ pub enum EmbedderEvent {
     ChangeBrowserVisibility(TopLevelBrowsingContextId, bool),
     /// Virtual keyboard was dismissed
     IMEDismissed,
+    /// Pause the compositor when underlying native surface is lost
+    PauseCompositor,
```

Comment:
It might be worth renaming these to make it clearer what these are handling. Perhaps something like:

```
InvalidateNativeSurface,
```

Reply:
FWIW, I wasn't sure about the correctness of the current way of handling this and suspected that eventually this will include more work such as pausing the script thread, animation etc in addition to binding/unbinding surface. But I'll change it to the suggested names for now and we can rename it later.